evalml/exceptions/exceptions.py

```diff
@@ -143,3 +143,19 @@ class PartialDependenceError(ValueError):
     def __init__(self, message, code):
         self.code = code
         super().__init__(message)
+
+
+class PipelineErrorCodeEnum(Enum):
+    """Enum identifying the type of error encountered while applying a pipeline."""
+
+    PREDICT_INPUT_SCHEMA_UNEQUAL = "predict_input_schema_unequal"
+    """predict_input_schema_unequal"""
+
+
+class PipelineError(ValueError):
```

Comment:
Nit: can we doc-string this and the PartialDependenceError above while we're here?